src/include/network/complex.rb

```diff
@@ -401,18 +401,17 @@ def DeviceStatus(devtype, devname, devmap)
     # @param [Hash] devmap device map
     # @return textual device protocol
     def DeviceProtocol(devmap)
-      devmap = deep_copy(devmap)
-      if Ops.get_string(devmap, "STARTMODE", "") == "managed"
+      if devmap["STARTMODE"] == "managed"
         # Abbreviation for "The interface is Managed by NetworkManager"
         return _("Managed")
       end
-      ip = Ops.get_string(devmap, "BOOTPROTO", "static")
-      ip = if ip.nil? || ip == "" || ip == "static"
-        Ops.get_string(devmap, "IPADDR", "")
+      bootproto = devmap["BOOTPROTO"] || "static"
+
+      if bootproto.empty? || bootproto == "static"
+        devmap["IPADDR"]
```

Comment:
In case of `nil` `bootproto` is set to `"static"` as you can see in line 408. Now we have a unit test for `DeviceProtocol` method and we can see that for `nil`, `""` and `"static"` we return the IP address.